src/webmock/core_ext.cr

```diff
@@ -3,6 +3,10 @@ require "http/client"
 class HTTP::Request
   getter scheme
   setter scheme : String = "http"
+
+  def full_uri
```

Comment:
Is this extension really needed? I seems to be that it should belong to the `matches_uri?` method or at least the to the `WebMock::Stub` class.

Reply:
In terms of use cases, yes only matches_uri? method wants this in this change.

But in terms of responsibility, I think this is perfectly a behavior of HTTP::Request itself, since it's result is 100% decided by the state of the single object.

Reply:
I see the point, I guess it's a matter of taste and how I would've written.